terraform/groups/lambda/module-lambda-roles/main.tf

```diff
@@ -0,0 +1,96 @@
+# ------------------------------------------------------------------------------
+# Policy Documents
+# ------------------------------------------------------------------------------
+data "aws_iam_policy_document" "cdn_analyser_trust" {
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "sts:AssumeRole",
+    ]
+
+    principals {
+      type = "Service"
+
+      identifiers = [
+        "lambda.amazonaws.com",
+      ]
+    }
+  }
+}
+
+data "aws_iam_policy_document" "cdn_analyser_execution" {
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "logs:DescribeQueries",
+      "logs:GetLogRecord",
+      "s3:PutAccountPublicAccessBlock",
+      "logs:PutDestinationPolicy",
+      "logs:StopQuery",
+      "logs:TestMetricFilter",
+      "logs:DeleteDestination",
+      "logs:CreateLogGroup",
+      "logs:GetLogDelivery",
+      "logs:ListLogDeliveries",
+      "logs:CreateLogDelivery",
+      "logs:DeleteResourcePolicy",
+      "logs:PutResourcePolicy",
+      "logs:DescribeExportTasks",
+      "s3:GetAccountPublicAccessBlock",
+      "logs:GetQueryResults",
+      "s3:ListAllMyBuckets",
+      "logs:UpdateLogDelivery",
+      "logs:CancelExportTask",
+      "logs:DeleteLogDelivery",
+      "s3:HeadBucket",
+      "s3:GetObject",
+      "logs:PutDestination",
+      "logs:DescribeResourcePolicies",
+      "logs:DescribeDestinations",
+      "ec2:CreateNetworkInterface",
+      "ec2:DescribeNetworkInterfaces",
+      "ec2:DeleteNetworkInterface"
```

Comment:
There's a lot going on in this block. I'd maybe look to split out by aws function